src/omnicore/rules.cpp

```diff
@@ -317,41 +339,45 @@ bool ActivateFeature(uint16_t featureId, int activationBlock, int transactionBlo
         (activationBlock > (transactionBlock + params.MAX_ACTIVATION_BLOCKS))) {
-        PrintToLog("Feature activation of ID %d refused due to notice checks\n", featureId);
-        return false;
+            PrintToLog("Feature activation of ID %d refused due to notice checks\n", featureId);
+            return false;
     }
 
     // check feature is recognized and activation is successful
+    std::string featureName;
+    bool supported = OMNICORE_VERSION >= minClientVersion;
     switch (featureId) {
         case FEATURE_CLASS_C:
             MutableConsensusParams().NULLDATA_BLOCK = activationBlock;
-            PrintToLog("Feature activation of ID %d succeeded. "
-                       "Class C transaction encoding is going to be enabled at block %d.\n",
-                       featureId, params.NULLDATA_BLOCK);
-            return true;
+            featureName = "Class C transaction encoding";
+        break;
         case FEATURE_METADEX:
             MutableConsensusParams().MSC_METADEX_BLOCK = activationBlock;
-            PrintToLog("Feature activation of ID %d succeeded. "
-                       "The distributed token exchange is going to be enabled at block %d.\n",
-                       featureId, params.MSC_METADEX_BLOCK);
-            return true;
+            featureName = "Distributed Meta Token Exchange";
+        break;
         case FEATURE_BETTING:
             MutableConsensusParams().MSC_BET_BLOCK = activationBlock;
-            PrintToLog("Feature activation of ID %d succeeded. "
-                       "Bet transactions are going to be enabled at block %d.\n",
-                       featureId, params.MSC_BET_BLOCK);
-            return true;
+            featureName = "Bet transactions";
+        break;
         case FEATURE_GRANTEFFECTS:
             MutableConsensusParams().GRANTEFFECTS_FEATURE_BLOCK = activationBlock;
-            PrintToLog("Feature activation of ID %d succeeded. "
-                       "The potential side effect of crowdsale participations, when "
-                       "granting tokens, is going to be disabled at block %d.\n",
-                       featureId, params.GRANTEFFECTS_FEATURE_BLOCK);
-            return true;
+            featureName = "Remove grant side effects";
+        break;
+        default:
+            featureName = "Unknown feature";
+            supported = false;
+        break;
     }
-
-    PrintToLog("Feature activation of id %d refused due to unknown feature ID\n", featureId);
-    return false;
+    PrintToLog("Feature activation of ID %d processed. %s will be enabled at block %d.\n", featureId, featureName, activationBlock);
+    AddPendingActivation(featureId, activationBlock, minClientVersion, featureName);
+    AddAlert("omnicore", ALERT_BLOCK_EXPIRY, activationBlock, strprintf("Feature %d ('%s') will go live at block %d", featureId, featureName, activationBlock));
+    if (!supported) {
```

Comment:
Whenever I encounter an unsupported activation, the client freezes. To be more specific:
```
# fresh start, regtest mode with -omnialertallowsender=any
setgenerate true 101
# send some coins to a new address
omni_sendactivation mr49RtQEnNz1eukzJxxEv8hGRBu5cAjwp3 9 110 0
setgenerate true 1
# freeze ...
```
Or:
```
# fresh start, regtest mode with -omnialertallowsender=any
setgenerate true 101
# send some coins to a new address
omni_sendactivation mr49RtQEnNz1eukzJxxEv8hGRBu5cAjwp3 1 110 999999999
setgenerate true 1
# freeze ...
```
Valid activations don't cause a freeze, and go through until the end (-> actual activation -> no longer a notice after 1024 in the UI).

Restarting or reparsing doesn't help, and once the unsupported activation is processed, it ends, both via `omnicored` and `omnicore-qt`.

Log:
```
2015-08-17 00:57:17 Class C transaction detected: e7b58c7434a2a6f4ccb6a917bc28a0660e0522a246346e1964d18619ffb1bf8a parsed to fffffffe00060000006e3b9ac9ff at vout 0
2015-08-17 00:57:17 single_pkt: fffffffe00060000006e3b9ac9ff
2015-08-17 00:57:17 	------------------------------
2015-08-17 00:57:17 	         version: 65535, class C
2015-08-17 00:57:17 	            type: 65534 (Feature Activation)
2015-08-17 00:57:17 	      feature id: 6
2015-08-17 00:57:17 	activation block: 110
2015-08-17 00:57:17 	 minimum version: 999999999
2015-08-17 00:57:17 	          sender: mr5Q4dPahLvfdmyouemQHFgQeRcMP8GXx3
2015-08-17 00:57:17 	      authorized: true
2015-08-17 00:57:17 Feature activation requested (ID 6 to go active as of block: 110)
2015-08-17 00:57:17 Feature activation of ID 6 processed. Unknown feature will be enabled at block 110.
2015-08-17 00:57:17 New alert added: omnicore, 1, 110, Feature 6 ('Unknown feature') will go live at block 110
2015-08-17 00:57:17 WARNING!!! AS OF BLOCK 110 THIS CLIENT WILL BE OUT OF CONSENSUS AND WILL AUTOMATICALLY SHUTDOWN.
2015-08-17 00:57:17 New alert added: omnicore, 0, 110, Your client must be updated and will shutdown at block 110 (unsupported feature 6 ('Unknown feature') activated)

2015-08-17 00:57:17 recordTX(e7b58c7434a2a6f4ccb6a917bc28a0660e0522a246346e1964d18619ffb1bf8a, valid=YES, block= 103, type= 65534, value= 0)
2015-08-17 00:57:17 recordTX(): OK, line 3389, file: omnicore/omnicore.cpp
2015-08-17 00:57:17 devmsc=4181831959914, exodus_prev=4181831482244, exodus_delta=477670
2015-08-17 00:57:17 update_tally_map(mpexoDuSkGGqvqrkrjiFng38QPkJQVFyqv, 1=0x1, +477670, ttype=0): before=4181831482244, after=4181831959914
2015-08-17 00:57:17 devmsc for block 103: 477670, Exodus balance: 41818.31959914
```

Edit: updated the log with `-omnidebug=all`. It looks like the freeze happens, when the *new* block is processed.